examples/gitops/argocd-multi-cluster/hub-cluster/main.tf

```diff
@@ -0,0 +1,372 @@
+provider "aws" {
+  region  = local.region
+  profile = local.hub_profile
+}
+
+provider "kubernetes" {
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = try(base64decode(module.eks.cluster_certificate_authority_data), "")
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    args        = ["eks", "get-token", "--cluster-name", local.name, "--region", local.region, "--profile", local.hub_profile]
+    command     = "aws"
+  }
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = module.eks.cluster_endpoint
+    cluster_ca_certificate = try(base64decode(module.eks.cluster_certificate_authority_data), "")
+    exec {
+      api_version = "client.authentication.k8s.io/v1beta1"
+      args        = ["eks", "get-token", "--cluster-name", local.name, "--region", local.region, "--profile", local.hub_profile]
+      command     = "aws"
+    }
+  }
+}
+
+provider "kubectl" {
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = try(base64decode(module.eks.cluster_certificate_authority_data), "")
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    args        = ["eks", "get-token", "--cluster-name", local.name, "--region", local.region, "--profile", local.hub_profile]
+    command     = "aws"
+  }
+  load_config_file  = false
+  apply_retry_count = 15
+}
+
+# To get the hosted zone to be use in argocd domain
+data "aws_route53_zone" "domain_name" {
+  count        = local.enable_ingress ? 1 : 0
+  name         = local.domain_name
+  private_zone = local.domain_private_zone
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_availability_zones" "available" {}
+
+locals {
+  name        = var.hub_cluster_name
+  domain_name = var.domain_name
+
+  cluster_version = "1.24"
+
+  instance_type = "m5.large"
+
+  vpc_cidr  = "10.0.0.0/16"
+  azs_count = 3
+  azs       = slice(data.aws_availability_zones.available.names, 0, local.azs_count)
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/aws-ia/terraform-aws-eks-blueprints"
+  }
+
+  argocd_namespace         = "argocd"
+  argocd_namespaces_prefix = var.argocd_namespaces_prefix
+
+  # Ingress Config
+  enable_ingress      = var.enable_ingress
```

Comment:
lets try to avoid re-wrapping variables as locals. a variable (`var.xxx`) is intended to expose a configurable interface, whereas a local is not (think of it as a private variable)